packages/contracts/src/contracts/current/protocol/Exchange/MixinWrapperFunctions.sol

```diff
@@ -16,281 +16,259 @@
+                    signatures[i]
+                )
+            );
+            if (totalTakerTokenFilledAmount == takerTokenFillAmount) {
+                break;
+            }
         }
         return totalTakerTokenFilledAmount;
     }
 
     /// @dev Synchronously executes multiple calls of fillOrderNoThrow in a single transaction until total takerTokenFillAmount filled.
-    /// @param orderAddresses Array of address arrays containing individual order addresses.
-    /// @param orderValues Array of uint256 arrays containing individual order values.
+    /// @param orders Array of orders.
     /// @param takerTokenFillAmount Desired total amount of takerToken to fill in orders.
     /// @param signatures Maker's signatures of the orders.
     /// @return Total amount of takerTokenFillAmount filled in orders.
     function marketFillOrdersNoThrow(
-        address[5][] orderAddresses,
-        uint256[6][] orderValues,
+        Order[] orders,
         uint256 takerTokenFillAmount,
         bytes[] signatures)
         public
         returns (uint256 totalTakerTokenFilledAmount)
     {
-        for (uint256 i = 0; i < orderAddresses.length; i++) {
-            require(orderAddresses[i][3] == orderAddresses[0][3]); // takerToken must be the same for each order
-            var (, takerTokenFilledAmount) = fillOrderNoThrow(
-                orderAddresses[i],
-                orderValues[i],
-                safeSub(takerTokenFillAmount, totalTakerTokenFilledAmount),
-                signatures[i]
+        for (uint256 i = 0; i < orders.length; i++) {
+            require(orders[i].takerTokenAddress == orders[0].takerTokenAddress);
+            totalTakerTokenFilledAmount = safeAdd(
+                totalTakerTokenFilledAmount,
+                fillOrderNoThrow(
+                    orders[i],
+                    safeSub(takerTokenFillAmount, totalTakerTokenFilledAmount),
+                    signatures[i]
+                )
             );
-            totalTakerTokenFilledAmount = safeAdd(totalTakerTokenFilledAmount, takerTokenFilledAmount);
-            if (totalTakerTokenFilledAmount == takerTokenFillAmount) break;
+            if (totalTakerTokenFilledAmount == takerTokenFillAmount) {
+                break;
+            }
         }
         return totalTakerTokenFilledAmount;
     }
 
     /// @dev Synchronously cancels multiple orders in a single transaction.
-    /// @param orderAddresses Array of address arrays containing individual order addresses.
-    /// @param orderValues Array of uint256 arrays containing individual order values.
+    /// @param orders Array of orders.
     /// @param takerTokenCancelAmounts Array of desired amounts of takerToken to cancel in orders.
     function batchCancelOrders(
-        address[5][] orderAddresses,
-        uint256[6][] orderValues,
+        Order[] orders,
         uint256[] takerTokenCancelAmounts)
-        external
+        public
```

Comment:
Is this to allow it to be called from other smart contracts?

Reply:
Yep. External visibility doesn't work with AbiV2 yet.

Reply:
> Is this to allow it to be called from other smart contracts?

`external` means it can be called from transactions and other contracts, but not internally by this contract. (Though it can still do so through a `CALL`). It would not make sense if there wasn't a slight performance bonus to it.